apps/portal/js/assets.js

```diff
@@ -147,6 +149,48 @@ $(function () {
     };
 
     /**
+     * To get the gadget usage information for a particular gadget id
+     * @param gadgetId Id of the gadget
+     */
+    var getGadgetUsageInfo = function(gadgetId) {
+        var gadgetUsageData;
+        $.ajax({
+            url: DATABASE_API + '/' + gadgetId + '?task=usage',
+            method: "POST",
+            async: false,
+            contentType: 'application/json',
+            success: function (data) {
+                console.log("Successfully retrieved gadget usage information");
+                gadgetUsageData = JSON.parse(data);
+            },
+            error: function (xhr, message) {
+                console.log("something went wrong while retrieving the usage data. " + message);
```

Comment:
We need to reconsider whether we are going to log the error in browser or show it to user in some other way.